app/src/main/java/xyz/lilsus/papp/data/repository/WalletRepositoryImpl.kt

```diff
@@ -1,15 +1,65 @@
 package xyz.lilsus.papp.data.repository
 
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.core.DataStoreFactory
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.first
 import xyz.lilsus.papp.common.Invoice
+import xyz.lilsus.papp.data.WalletConfigSerializer
 import xyz.lilsus.papp.data.remote.blink.BlinkWalletApi
 import xyz.lilsus.papp.domain.model.SendPaymentResult
+import xyz.lilsus.papp.domain.repository.WalletApi
 import xyz.lilsus.papp.domain.repository.WalletRepository
+import xyz.lilsus.papp.proto.wallet_config.WalletConfig
+import xyz.lilsus.papp.proto.wallet_config.WalletType
+import java.io.File
+import java.io.IOException
+
+class WalletRepositoryImpl(
+    context: Context
+) : WalletRepository {
+
+    private val walletConfigStore: DataStore<WalletConfig> = DataStoreFactory.create(
+        serializer = WalletConfigSerializer,
+        produceFile = { File(context.filesDir, "wallet_config.pb") }
+    )
+
+    private var walletClient: WalletApi? = null
+    private var walletConfig: WalletConfig? = null
+
+    private suspend fun ensureClientInitialized() {
```

Comment:
I don't like this 'side effect' style guarantee, rather access the client behind a getter that ensures it's initialized.